lib/rpmsg/rpmsg.c

```diff
@@ -150,24 +150,26 @@ struct rpmsg_endpoint *rpmsg_get_endpoint(struct rpmsg_device *rdev,
 	struct rpmsg_endpoint *ept;
 
 	metal_list_for_each(&rdev->endpoints, node) {
-		int name_match = 1;
+		int name_match;
 
 		ept = metal_container_of(node, struct rpmsg_endpoint, node);
+		/* try to get by local address only */
 		if (addr != RPMSG_ADDR_ANY && ept->addr == addr)
 			return ept;
+		/* try to find match on local end remote address */
+		if (addr == ept->addr && dest_addr == ept->dest_addr)
+			return ept;
+		/* else use name service and destination address */
 		if (name)
 			name_match = !strncmp(ept->name, name,
 					      sizeof(ept->name));
-		if (dest_addr == RPMSG_ADDR_ANY &&
-		    ept->addr == RPMSG_ADDR_ANY &&
-		    name_match)
-			return ept;
-		if (addr == RPMSG_ADDR_ANY &&
-		    ept->dest_addr == RPMSG_ADDR_ANY &&
-		    name && name_match)
+		if (!name || !name_match)
```

Comment:
looks like name_match may be used as not initialized. As in the previous line name_match is only set when name is not NULL.

Reply:
No issue as name is tested first.  so test here is equivalent to
if( !name) 
     continue
else if (!name_match)
     continue

Reply:
you're right. Should be fine. Just not sure if there will be compilation warning. Will know when I compile with the patch.

Reply:
when I compile it i did got warning saying that it may be used uninitialized, i will change to assign default value to 0 when declaring the variable.